enhancements/split_helm_ansible_into_their_repos.md

```diff
@@ -0,0 +1,187 @@
+
+### Goals
+
+- The SDK CLI tool repository has just the code implementation, which is part of its domain of responsibility. 
+- A Helm Operator repository with all code implementation which is specific to it and is part of its domain of responsibility
+- An Ansible Operator repository with all code implementation which is specific to it and is part of its domain of responsibility
+- An SDK Operator Lib which contains the standard code that should be used to generate all types of Operators. (e.g. utils for releases which might be expected for all, templates and the implementation of the plugins which can be used for any language/type of operator)
+- Ansible and Helm repository have their godocs generated automatically 
+- Ansible and Helm repository have their CI, makefile targets and helpers and utilities required to contribute to them individually and test the projects scaffolded for these types. 
+- Helm/Ansible binaries should be created/released from their repos and no longer from SDK CLI. 
+
+### Non-Goals
+
+- Apply the changes for new Helm based-operator implementation in the [joelanford/helm-operator](https://github.com/joelanford/helm-operator) repository.
+- Steps required to achieve the goal of the types projects have their own docs subdomains `ansible.operatorframework.io` and `helm.operatorframework.io`  
+ 
+## Proposal
+
+### I am as a mantainer and/or community contributor would like to 
+- be able to check the CI make its work in a reasonable time
+- be able to quickly identify what part of the domain of responsibility of each type is and what is shared between them. 
+- be able to create new Operator types using the SDK Operator lib easily
+- be able to easily maintain new Operator types by updating its lib and not caring about changes made in the common base.
+- be able to patch/update just one place when we bump new upstream versions instead of needing to update its duplications across all projects.   
+- be able to provide releases according to the lifecycle of each type individually
+
+### I am as a user would like to 
+- upgrade/migrate my projects just only if a new release contains changes for the specific type of the used project 
+- develop my own types base on the helpers provided by the SDK Operator lib 
+- have better visibility about what was changed in the SDK releases and per type
+- not have more than one standard/command or workflow to achieve the same need with different types of project
+- not need to learn how to configure and working on with the operator projects built with SDK CLI tool from scratch when I need to create operators with different types.   
+- use the Helm/Ansible pkg implementation to create my customized helm/ansible based-operator images 
+
+### Implementation Details/Notes/Constraints 
+
+- The code implementation which will be moved from SDK repository to the new types repositories needs to keep their GitHub history tracked. 
```

Comment:
This is a pretty significant constraint. Can you expand upon how this will be done?

Reply:
I thought that @jmrodri could find a solution to do it for operator-lib. However, I just checked its logs and shows that not. Removing it 👍